client/signup/main.jsx

```diff
@@ -304,28 +304,25 @@ const Signup = React.createClass( {
 				window.scrollBy( 0, -10 );
 			} else {
 				this.setState( { scrolling: false } );
-				this.loadNextStep();
+				clearInterval( this.windowScroller );
```

Comment:
Sorry if i missed the obvious, but I have a few questions:

1. Why the interval ID needs to be an object property? ( `this.windowScroller` -vs- `const windowScrollIntervalId`

2. What you think about separating the scrolling for the redirecting to next step?
```js
// Animate the scroll
const scrollDonePromise = new Promise( resolve => {
	const scrollIntervalId = setInterval( () => {
		if ( window.pageYOffset > 0 ) {
  			window.scrollBy( 0, -10 );
  		} else {
  			clearInterval( scrollIntervalId );
  			resolve();
  		}
	} )
} );

// Redirect to next step
scrollDonePromise.then( () =>
	this.isEveryStepSubmitted()
		? this.goToFirstInvalidStep()
		: page( utils.getStepUrl( this.props.flowName, stepName, stepSectionName, this.props.locale ) );
);
```